File: 58-length-of-last-word/length-of-last-word.c
#include <string.h>

int lengthOfLastWord(char *s) {
    int len = strlen(s);
    while (len > 0 && s[len - 1] == ' ') len--; 
    int result = 0;
    for (int i = len - 1; i >= 0; i--) {
        if (s[i] == ' ') {
            return result;
        }
        result++;
    }
    return result;
}

File: 58-length-of-last-word/length-of-last-word.go
func lengthOfLastWord(s string) int {
    s = strings.TrimSpace(s)
    for i := len(s) - 1; i >= 0; i-- {
        if s[i] == ' ' {
            return len(s[i+1:])
        }
    }
    return len(s)
}
